### ISSUE TITLE
Front end bug with _mm_mask_add_ph

### ISSUE THREAD

**Issue by AgnerF** [llvm/llvm-project#55635]
This front end bug appears with the _mm_mask_add_ph intrinsic. The bug disappears when the -ffp-model=strict option is off. A minimal test case doesn't cause the error.

[testbench4-13c79a.zip](https://github.com/llvm/llvm-project/files/8749636/testbench4-13c79a.zip)

**Comment by Endilll** [comment]
Appears to be fixed on post-17 trunk: https://godbolt.org/z/of6v5rcjP
Reduced by C-Reduce:
```cpp
typedef _Float16 __m128h __attribute__((__vector_size__(16), __aligned__0));
class Vec8h {
public:
  Vec8h;
  operator __m128h() const;
};
void operator>(Vec8h const a, Vec8h const b) {
  __attribute__((
      __vector_size__(8 * sizeof(_Float16)))) _Float16 __trans_tmp_1 = a,
                                                       __trans_tmp_2 = b;
  __builtin_ia32_cmpph128_mask(__trans_tmp_1, __trans_tmp_2, 0, 0 - 1);
}
```
